extra_data/auxiliary.py

```diff
@@ -0,0 +1,208 @@
+
+from itertools import groupby
+import fnmatch
+import re
+
+from . import DataCollection, H5File, FileAccess, SourceData, SourceNameError
+
+
+class AuxiliaryIndexer:
+    """Enables item access to auxiliary sources."""
+
+    def __init__(self, train_ids, files, is_single_run, inc_suspect_trains):
+        # Resolve any voview files if present as they lack the auxiliary
+        # sources present in their source files.
+        files = self._resolve_voview_files(files)
+
+        # {(source, section) -> {aggregator -> files}}
+        files_by_sources = dict()
+
+        for fa in files:
+            aggregator = fa.aggregator
+
+            for source in fa.reduction_sources:
+                files_by_sources.setdefault((source, 'REDUCTION'), dict()) \
+                    .setdefault(aggregator, []).append(fa)
+
+            for source in fa.errata_sources:
+                files_by_sources.setdefault((source, 'ERRATA'), dict()) \
+                    .setdefault(aggregator, []).append(fa)
+
+        self._sources_data = dict()
+
+        for (src, section), files_by_aggregator in files_by_sources.items():
+            with_prefix = True
+
+            if len(files_by_aggregator) == 1:
+                # Don't prefix aggregator if there is exactly one
+                # aggregator with this source and it's the suffix of an
+                # actual source in the same aggregator.
+                for fa in next(iter(files_by_aggregator.values())):
+                    for real_src in fa.all_sources:
+                        if src.endswith(real_src):
+                            with_prefix = False
+                            break
+
+            self._sources_data.update({
+                f'{aggregator}@{src}' if with_prefix else src: SourceData(
+                    src, sel_keys=None, train_ids=train_ids,
```

Comment:
I just realized while looking at the tests that with prefixing, we generate `SourceData` objects for which `src != run.auxiliary[src].source`. Do we consider this a problem?

The alternative could be to go back to explicit aggregators while indexing. With the mechanism now in place to distinguish sources in need of prefixing and not, we could probably make `__getitem__` clever enough to accept `[[da], src, [key]]` reliably.